src/utils.rs

```diff
@@ -587,3 +587,101 @@ pub fn get_rsa_public(restricted: bool, decrypt: bool, sign: bool, key_bits: u16
         .build()
         .unwrap() // should not fail as we control the params
 }
+
+#[derive(Clone, Copy)]
+pub enum Hierarchy {
+    Null,
+    Owner,
+    Platform,
+    Endorsement,
+}
+
+impl Hierarchy {
+    pub fn esys_rh(self) -> TPMI_RH_HIERARCHY {
+        match self {
+            Hierarchy::Null => ESYS_TR_RH_NULL,
+            Hierarchy::Owner => ESYS_TR_RH_OWNER,
+            Hierarchy::Platform => ESYS_TR_RH_PLATFORM,
+            Hierarchy::Endorsement => ESYS_TR_RH_ENDORSEMENT,
+        }
+    }
+
+    pub fn rh(self) -> TPM2_RH {
+        match self {
+            Hierarchy::Null => TPM2_RH_NULL,
+            Hierarchy::Owner => TPM2_RH_OWNER,
+            Hierarchy::Platform => TPM2_RH_PLATFORM,
+            Hierarchy::Endorsement => TPM2_RH_ENDORSEMENT,
+        }
+    }
+}
+
+impl TryFrom<TPM2_HANDLE> for Hierarchy {
+    type Error = Error;
+
+    fn try_from(handle: TPM2_HANDLE) -> Result<Self> {
+        match handle {
+            TPM2_RH_NULL | ESYS_TR_RH_NULL => Ok(Hierarchy::Null),
+            TPM2_RH_OWNER | ESYS_TR_RH_OWNER => Ok(Hierarchy::Owner),
+            TPM2_RH_PLATFORM | ESYS_TR_RH_PLATFORM => Ok(Hierarchy::Platform),
+            TPM2_RH_ENDORSEMENT | ESYS_TR_RH_ENDORSEMENT => Ok(Hierarchy::Endorsement),
+            _ => Err(Error::local_error(WrapperErrorKind::InconsistentParams)),
+        }
+    }
+}
+
+pub struct TpmtTkVerified {
+    hierarchy: Hierarchy,
+    digest: Vec<u8>,
+}
+
+impl TpmtTkVerified {
+    pub fn hierarchy(&self) -> Hierarchy {
+        self.hierarchy
+    }
+
+    pub fn digest(&self) -> &[u8] {
+        &self.digest
+    }
+}
+
+impl TryFrom<TPMT_TK_VERIFIED> for TpmtTkVerified {
+    type Error = Error;
+
+    fn try_from(tss_verif: TPMT_TK_VERIFIED) -> Result<Self> {
+        if tss_verif.tag != TPM2_ST_VERIFIED {
+            return Err(Error::local_error(WrapperErrorKind::InconsistentParams));
+        }
+
+        let len = tss_verif.digest.size.try_into().unwrap();
```

Comment:
`unwrap` 😨 Maybe a comment would be nice to be consistent with our recent changes?

Reply:
Good spot, I'd forgotten

Reply:
Actually, that one's `u16` -> `usize` which can happen losslessly, so I'll just add a comment to the `unwrap` on line 682.